x-pack/plugins/observability_solution/logs_explorer/public/state_machines/logs_explorer_controller/src/services/discover_service.ts

```diff
@@ -106,3 +109,33 @@ export const updateDiscoverAppStateFromContext: ActionFunction<
 
   context.discoverStateContainer.appState.update(getDiscoverAppStateFromContext(context));
 };
+
+export const redirectToDiscoverAction =
+  (
+    discover: DiscoverStart
+  ): ActionFunction<LogsExplorerControllerContext, LogsExplorerControllerEvent> =>
+  (context, event) => {
+    if (event.type === 'UPDATE_DATASET_SELECTION' && isDataViewSelection(event.data)) {
+      return redirectToDiscover({ context, datasetSelection: event.data, discover });
```

Comment:
The way the action is implemented (and how the selector acted before) would cause the page-wide side-effect of navigating to a different URL. Is that desirable when the logs explorer is used in an embedded fashion? Should we delegate that to the consumer?

If so, how do we do that? 🤔 Two options come to mind:

- a customization that customizes the action that happens here (like the one used for the ai assistant)
- emitting an event on the (yet unused) `event$` observable that the consumer can react to

What do you think?

Reply:
I see your point here and both the proposed solutions LGTM.
On the other hand, what do you think should be the fallback behaviour when the consumer doesn't apply the customization or doesn't react to the event? 

My concern is we might end up with a broken UX when the user clicks on a non-logs data view if the consumer embedding the LogsExplorer doesn't react to the emitted event in `$event` (assuming we go with the second option).

One option to keep the default behaviour without forcing the user to react to the emitted event, but still provide the flexibility to perform custom logic and prevent the default behaviour if necessary could be to provide something like a `preventDefault()` method as part of the emitted event, in a browser-fashion way.

What I mean is, in case we'll extend the set of events emitted by the controller, to have something like this:
```ts
export type LogsExplorerPublicEvent =
  | {
      type: 'REDIRECT_TO_DISCOVER';
      data: { context, eventPayload };
      preventDefault: () => void;
    }
  | {
      type: 'ANOTHER_EVENT';
      data: { ... };
      preventDefault: () => void;
    }
  | ...
```

This is just one possible approach, but it might be more convoluted than other options, what do you think?
Or did I misunderstand your suggestion and you meant something different? 😅

Reply:
Mmm I've been thinking more about the approach I proposed for preventing the default behaviour as we might have more subscribers on the same emitted event, which makes it impractical to decide whether the default behaviour should be skipped or not...
Adding a customization point would work, but it still leaves the discussion open about what happens if the consumer doesn't register the customization.

Reply:
Based on a offline conversation with Felix, we'll implement a customization point to delegate the redirect to the consumer in a shorted upcoming pull request.